docs/2-fork-accountability.md

```diff
@@ -0,0 +1,191 @@
+In this case, the monitor would be able to complete the accountability algorithm correctly (respecting both the completeness and accuracy properties) as soon as at least *f + 1* faulty processes will be detected and by relying on the fact that at least *f + 1* correct processes will send correct message logs.
+
+## Algorithm design
+ 
+Assuming the small modification in the Tendermint consensus algorothm described above, these are the high-level steps carried out by the monitor to run the asynchronous accountability algorithm:
+
+1. Upon a fork, monitor requests message logs from all the validators
+
+2. Monitor runs the accountability algorithm to detect faulty processes upon receiving at least *f + 1* message logs and at every new message logs received when there are at least *f + 1* message logs 
+
+3. While running the accountability algorithm, the monitor analyzes the received logs: for every message *m* that has been received by some other process but is not present in the sender's message log, the monitor attaches *m* to the message log of the sender of *m*
+
+4. For each validator, the monitor scans the received message logs and the inferred message logs and determines for each process whether it is faulty or not by analyzing the history of sent messages and transitions
+
+5. When the monitor detects at least *f + 1* faulty processes, the algorithm completes
+
+Termination is guaranteed by the fact that correct processes will send their messages logs and their message logs will be correct (no sent message will be missing). 
+If this condition does not hold, the algorithm would not be able to identify correctly all the faulty processes that led to a fork. However, correct processes have no reason to misbehave or send invalid message logs. 
+
+## Notes
+
+- As some readers might have noticed, the fork accountability problem can be easily solved by simply piggybacking justifications in every single message. 
+However, this solution would be inefficient because it considerably increases the amount of information stored and exchanged by processes. The scope of the discussion is to solve the problem efficiently with a minimal amount of changes required on the Tendermint consensus protocol and that is the reason why we do not consider and analyze this possible solution.   
+
+- By excluding the piggybacking solution of the previous point, the fork accountability cannot be solved when the number of faulty proccesses is greater than *2f*: in this case, faulty processes basically control the system and nothing can be done to solve the problem.
+
+- The accountability algorithm is able to also catch other faulty behaviours that did not necessarily lead to a fork - this is a positive side effect of the accountability algorithm described above.
+
+- The monitor needs to analyze the message logs from the round of the first decision to the round of the second decision. Previous message history is not necessary to make the algorithm work. 
```

Comment:
This is true from the perspective of the monitor.
However, participants of the consensus can not know when some process decided (at least in Tendermint we have right now).

Reply:
True, modified the sentence to make this clear.